src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
```

Comment:
```suggestion
        array = new Object[DEFAULT_CAPACITY];
```
0 is a default value for int
let's not use `this` where we don't need to - https://stackoverflow.com/a/2411283